.github/workflows/ci.yml

```diff
@@ -70,68 +70,97 @@ jobs:
+      kubernetesModes: ${{ steps.selective-checks.outputs.kubernetes-modes }}
+      defaultKubernetesMode: ${{ steps.selective-checks.outputs.default-kubernetes-mode }}
+      postgresVersions: ${{ steps.selective-checks.outputs.postgres-versions }}
+      defaultPostgresVersion: ${{ steps.selective-checks.outputs.default-postgres-version }}
+      mysqlVersions: ${{ steps.selective-checks.outputs.mysql-versions }}
+      defaultMySQLVersion: ${{ steps.selective-checks.outputs.default-mysql-version }}
+      helmVersions: ${{ steps.selective-checks.outputs.helm-versions }}
+      defaultHelmVersion: ${{ steps.selective-checks.outputs.default-helm-version }}
+      kindVersions: ${{ steps.selective-checks.outputs.kind-versions }}
+      defaultKindVersion: ${{ steps.selective-checks.outputs.default-kind-version }}
+      testTypes: ${{ steps.selective-checks.outputs.test-types }}
+      postgresExclude: ${{ steps.selective-checks.outputs.postgres-exclude }}
+      mysqlExclude: ${{ steps.selective-checks.outputs.mysql-exclude }}
+      sqliteExclude: ${{ steps.selective-checks.outputs.sqlite-exclude }}
+      kubernetesExclude: ${{ steps.selective-checks.outputs.kubernetes-exclude }}
+      run-tests: ${{ steps.selective-checks.outputs.run-tests }}
+      run-kubernetes-tests: ${{ steps.selective-checks.outputs.run-kubernetes-tests }}
+      basic-checks-only: ${{ steps.selective-checks.outputs.basic-checks-only }}
+      needs-helm-tests: ${{ steps.selective-checks.outputs.needs-helm-tests }}
+      needs-api-tests: ${{ steps.selective-checks.outputs.needs-api-tests }}
     if: github.repository == 'apache/airflow' || github.event_name != 'schedule'
     steps:
-      - name: "Checkout ${{ github.ref }} ( ${{ github.sha }} )"
+      # First fetch the sha of merge commit in case it is pull request so that we can
+      # Run selective tests
+      - name: >
+          Fetch merge commit ${{ github.ref }} ( ${{ github.sha }}:
+          merge_commit ${{ github.event.pull_request.merge_commit_sha }} )"
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.merge_commit_sha }}
+          fetch-depth: 2
+        if: github.event_name  == 'pull_request'
+      - name: "Checkout ${{ github.ref }} ( ${{ github.sha }} : merge commit ${{ github.merge_commit_sha }} )"
         uses: actions/checkout@v2
       - name: >
           Event: ${{ github.event_name }}
           Repo: ${{ github.repository }}
           Branch: ${{ github.head_ref }}
           Run id: ${{ github.run_id }}
           Sha: ${{ github.sha }}
+          Merge commit sha: ${{ github.merge_commit_sha }}
           Ref: ${{ github.ref }}
         run: printenv
       - name: Set wait for image
         id: wait-for-image
         run: |
-          if [[ ${GITHUB_REGISTRY_WAIT_FOR_IMAGE} == "true" ]]; then
+          if [[ ${GITHUB_REGISTRY_WAIT_FOR_IMAGE} == 'true' ]]; then
```

Comment:
```suggestion
          if [[ "${GITHUB_REGISTRY_WAIT_FOR_IMAGE}" == 'true' ]]; then
```

not strictly needed

Reply:
Same here  - there are possibly few hundreds of places all over our code where " no quote on the left side inside [[ ]] " is  followed. I'd prefer to keep it - it makes the code a bit more readable.